### ISSUE TITLE
qpzm - `PriceOracle.getPriceFromChainlink` is not validated properly.

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-ironbank-judging#383]
qpzm

medium

# `PriceOracle.getPriceFromChainlink` is not validated properly.

## Summary
`PriceOracle.getPriceFromChainlink` is not validated properly.

## Vulnerability Detail
1. `updatedAt` is not checked. https://docs.chain.link/data-feeds/api-reference#latestrounddata
2. `latestRoundData()` may be blocked by the multisigs in Chainlink.
```solidity
function getPriceFromChainlink(address base, address quote) internal view returns (uint256) {
    (, int256 price,,,) = registry.latestRoundData(base, quote);
    require(price > 0, "invalid price");

    // Extend the decimals to 1e18.
    return uint256(price) * 10 ** (18 - uint256(registry.decimals(base, quote)));
}
```

## Impact
`PriceOracle` contract may return stale price or revert.

## Code Snippet
https://github.com/sherlock-audit/2023-05-ironbank/blob/main/ib-v2/src/protocol/oracle/PriceOracle.sol#L66-L72

## Tool used

Manual Review

## Recommendation

```solidity
function getPriceFromChainlink(address base, address quote) internal view returns (uint256) {
    try registry.latestRoundData() returns (, int256 price,, uint256 updatedAt,) {
        require(price > 0, "invalid price");
        if (updatedAt < block.timestamp - 60 minutes) {
            // revert or pause the protocol
        }
    } catch {
        // revert or pause the protocol when Chainlink aggregator reverts
    }
```

## Reference
- https://github.com/liquity/dev/blob/main/packages/contracts/contracts/PriceFeed.sol#L511
- https://0xmacro.com/blog/how-to-consume-chainlink-price-feeds-safely/

Duplicate of #9
